### PR TITLE
Add batch file to support windows installation for fzf.exe

### PR DESCRIPTION
**Pull request by jiangjianshan**
To support following features:
1. To download the fzf.exe if it is not exist.
2. If the version of fzf.exe is not correct, also download the correct version.
3. The batch file will try to find the third-party uncompress tool, e.g 7-zip or WinRAR at Windows. If can't found it, they may be not installed or not has been inside PATH.

### CODE REVIEW

**Comment by jiangjianshan** [comment]
Why the travis CI build failed? I have test this .bat file at my laptop and running correctly.

File: install.bat [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+            echo Please intall 7-zip or WinRAR or put them into PATH if they has been installed!
+        ) else (
+            wget -O %temp% %~1 && winrar x %temp% && del /s /q %temp%
+        )
+    ) else (
+        wget -O %temp% %~1 && 7z -y x %temp% && del /s /q %temp%
+    )
+    exit /b 0
+)
+goto :eof
```
Comment (junegunn) [inline install.bat]: The code for wget and curl look more or less the same. Can we refactor the code and remove the repetition?

Comment (jiangjianshan) [reply install.bat]: @junegunn , I have simple the batch file by the commit ba2c4fda9d1d43d854f33ec2c3d98abe5cc77a58. The way to extract file was change to use one powershell command but not try to find out 7-zip or WinRAR on Windows. Please review the change on the function try_curl and try_wget.

**Comment by jiangjianshan** [comment]
@junegunn , today I have also using powershell script to implement the same feature as install.bat. The file install.ps1 is located at  https://gist.github.com/jiangjianshan/56934892ae101e3d81273f2324b93670. Which implement way you prefer? If you prefer to use powershell, you can review the file at gist.github.com also. If no, please ignore it.

File: install.bat (hunk 1 above)
Comment (junegunn) [reply install.bat]: > Which implement way you prefer, cmd or powershell?

I'm not a Windows user and have no prior experience with either. Which one do you prefer? :)

It looks like the PowerShell version doesn't require wget or curl, which definitely is a plus, right?

**Comment by jiangjianshan** [comment]
@junegunn , for me is prefer to use powershell also. I have just commit the file install.ps1 from https://gist.github.com/jiangjianshan/56934892ae101e3d81273f2324b93670, which commit id is f8e1d7202797a1ffad2102532d73fb68c25b787e. If there is no problem, it is recommend that you can merge it into the master.

**Comment by junegunn** [comment]
I've finally found a chance to test the script on my Windows VM (Windows Server 16), but I ran into this error.
```
C:\Users\Administrator\.vim\plugged\fzf>powershell -ExecutionPolicy Bypass
Windows PowerShell
Copyright (C) 2016 Microsoft Corporation. All rights reserved.

PS C:\Users\Administrator\.vim\plugged\fzf> [Threading.Thread]::CurrentThread.CurrentUICulture = 'en-US'; .\install.ps1
Downloading bin/fzf ...
  - Already exists
  - Checking fzf executable ... 0.19.0 != 0.20.0
Exception calling "DownloadFile" with "2" argument(s): "The request was aborted: Co
uld not create SSL/TLS secure channel."
At C:\Users\Administrator\.vim\plugged\fzf\install.ps1:57 char:5
+     (New-Object Net.WebClient).DownloadFile($url, $ExecutionContext.S ...
+     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [], MethodInvocationException
    + FullyQualifiedErrorId : WebException

binary_error=Failed to download with powershell : The term 'binary_error=Failed to
download with powershell' is not recognized as the name of a cmdlet, function, scri
pt file, or operable program. Check the spelling of the name, or if a path was incl
uded, verify that the path is correct and try again.
At C:\Users\Administrator\.vim\plugged\fzf\install.ps1:61 char:9
+         binary_error="Failed to download with powershell"
+         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : ObjectNotFound: (binary_error=Fa...with powershell:S
   tring) [], CommandNotFoundException
    + FullyQualifiedErrorId : CommandNotFoundException

For more information, see: https://github.com/junegunn/fzf
PS C:\Users\Administrator\.vim\plugged\fzf\bin>
```
Any idea what went wrong?

**Comment by jiangjianshan** [comment]
@junegunn ,  for the issue "Exception calling "DownloadFile" with "2" argument(s): "The request was aborted: Could not create SSL/TLS secure channel."
could you try this new file at https://gist.github.com/jiangjianshan/56934892ae101e3d81273f2324b93670?  The change are here:

**Comment by junegunn** [comment]
Thanks, I have noticed a few problems.

If `bin\fzf.exe` doesn't exist, the script fails. I had to change `$output=bin\fzf.exe --version 2>&1` to `$output=.\fzf.exe --version 2>&1` since the working directory is already changed to `bin` at that point.

Once I change the line, it succeeds to download the file:

```
C:\Users\Administrator\.vim\plugged\fzf>powershell -ExecutionPolicy Bypass -file install.ps1
Downloading bin/fzf ...
  - Checking fzf executable ... 0.20.0
1
For more information, see: https://github.com/junegunn/fzf
```

(By the way, we should get rid of `1` message)

However, re-running the script gives this error:

```
PS C:\Users\Administrator\.vim\plugged\fzf> [Threading.Thread]::CurrentThread.CurrentUICulture = 'en-US'; .\install.ps1
Downloading bin/fzf ...
  - Already exists
  - Checking fzf executable ... .\fzf.exe : The term '.\fzf.exe' is not recognized as the name of a cmdlet, function, script file, or operable program.
 Check the spelling of the name
, or if a path was included, verify that the path is correct and try again.
At C:\Users\Administrator\.vim\plugged\fzf\install.ps1:13 char:13
+     $output=.\fzf.exe --version 2>&1
+             ~~~~~~~~~
    + CategoryInfo          : ObjectNotFound: (.\fzf.exe:String) [], CommandNotFoundException
    + FullyQualifiedErrorId : CommandNotFoundException
 
Error:
  - Checking fzf executable ... 0.20.0
1
For more information, see: https://github.com/junegunn/fzf
```

**Comment by jiangjianshan** [comment]
@junegunn , this morning I have find out the solution for the issues you meet last night. you can use the new file install.ps1 at https://gist.github.com/jiangjianshan/56934892ae101e3d81273f2324b93670. 
Only two place I have modified. They are here:
image
and
image

**Comment by junegunn** [comment]
Thanks, the issues are fixed now.

One more thing: If I understood correctly, `Get-Location` returns the current working directory, but it's more desirable that we get the project root directory, so we can run the script from any directory. For example, if we do this:

```sh
cd ..
powershell -ExecutionPolicy Bypass -file fzf\install.ps1
```

The scripts downloads fzf.exe in `bin` directory under the current working directory.

The current Unix install script handles the case this way:

```sh
cd "$(dirname "${BASH_SOURCE[0]}")"
fzf_base=$(pwd)
```

**Comment by jiangjianshan** [comment]
@junegunn ,  you are right, the original implementation at Unix install script is that command gets the script's source file pathname, strips it to just the path portion, cd to that path, then uses pwd to return the (effectively) full path of the script. 
The Get-Location is an alias command for pwd at powershell.
image
This morning I have found an equivalent at powershell as Unix script do.  It is
image
I have put this new line modification at https://gist.github.com/jiangjianshan/56934892ae101e3d81273f2324b93670. 
Could you review it again? If no problem, I will commit so that you can merge it later.

**Comment by junegunn** [comment]
Thanks, now it works as expected. 👍

Additionally, I'd like to ask you to

1. remove `install.bat` as we don't want to maintain two scripts for doing the same thing
2. and change indentation of 4-spaces to 2-spaces to make it consistent with other scripts.

Thanks for your patience.

**Comment by jiangjianshan** [comment]
@junegunn , according to your recommend
image
I have just made a commit which id is 613903eb19e3975dbc6fe601701e95e21c1ae806.

**Comment by junegunn** [comment]
Looks great! Merged, thanks!

**Comment by jiangjianshan** [comment]
@junegunn , you are welcome. It is recommand that you can update the main page of https://github.com/junegunn/fzf. 
image

The one with red rectage line can be updated as, i.e.

if has('win32')
    Plug 'junegunn/fzf', { 'dir': '$HOME/.fzf', 'do': '!powershell -ExecutionPolicy Bypass -file install.ps1' }
else
    Plug 'junegunn/fzf', { 'dir': '~/.fzf', 'do': './install --all' }
endif

**Comment by junegunn** [comment]
@jiangjianshan Thanks for the input. I added `fzf#install()` function which should properly select the right command 👉 
 https://github.com/junegunn/fzf/commit/6c03571887cda7c0253c0a28439230cc8fcea3e6

**Pull request merged**